spec/unit/decorators/include_binding_service_instance_decorator_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+require 'decorators/include_binding_service_instance_decorator'
+
+module VCAP::CloudController
+  RSpec.describe IncludeBindingServiceInstanceDecorator do
+    subject(:decorator) { described_class }
+    let(:bindings) { [ServiceBinding.make, ServiceBinding.make, ServiceBinding.make] }
+    let(:instances) {
+      bindings.map { |b| Presenters::V3::ServiceInstancePresenter.new(b.service_instance).to_hash }
+    }
+
+    it 'decorates the given hash with service instances from bindings' do
+      dict = { foo: 'bar' }
+      hash = subject.decorate(dict, bindings)
+      expect(hash[:foo]).to eq('bar')
+      expect(hash[:included][:service_instances]).to match_array(instances)
+    end
+
+    it 'does not overwrite other included fields' do
+      dict = { foo: 'bar', included: { fruits: ['tomato', 'banana'] } }
+      hash = subject.decorate(dict, bindings)
+      expect(hash[:foo]).to eq('bar')
+      expect(hash[:included][:service_instances]).to match_array(instances)
+      expect(hash[:included][:fruits]).to match_array(['tomato', 'banana'])
+    end
+
```

Comment:
The implementation has code to de-duplicate service instances using `.uniq`, and I don't see a test that drives that implementation. Should we have one, or does the cost outweigh the benefit?

Reply:
It's indirectly (or even accidentally I'd say) tested in the request spec (there are 4 bindings originating from 2 instances). There's no harm in making it explicit and it should be simple enough to add in the decorator test.

Reply:
Interesting enough, the `.uniq` seems to be unnecessary to guarantee uniqueness in the returned result. I guess it's building a slightly better SQL query (`guid in ("a","b")` rather than `guid in ("a","b","b")`). I'm inclined to remove the `.uniq` and leave the new test... 🤔